### PR TITLE
Adding the /getAll route as well the status query parameter

### LINKED ISSUE #158: [Feature Request] - Create API route to return all support items
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Is your feature request related to a problem, or an open issue? Describe the problem, and if applicable, reference the issue.

We need a new endpoint to allow an admin to retrieve all support items from the UI

## Describe the solution you'd like

Create a new route nested within `api/supportRequest/` for `getAll`, which should return a list of all support request items. The method should default to only returning `Pending` items, however, a query parameter of `status` should be utilized to provide others if needed (list of values in `entities/supportRequest.ts`).

### PR DESCRIPTION
**Pull request by Ulfingrad**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

@msiddique933 and I Added a route called `getAll` which returns all support requests in the database. Additionally there is the option for a `status` parameter which will allow the user to fetch support requests based on their status. Notably we did not add testing, so that is something that needs to be implemented in the future.

## Related Issues

Closes #158 which was a feature request for adding a `/getAll` route for support requests. This includes `status` query parameter in which a user can search the support requests by their status.

### CODE REVIEW

**Comment by Ulfingrad** [comment]
Did I mess up when it comes to bypassing testing? I used `--no-verify` when committing. Was I supposed to put it somewhere else or something?

**Comment by johnkahn** [comment]
No verify will only skip tests on your local machine, but it'll still run it on github and won't let us merge it, so tests will need to be added for it to be mergable

@SpencerKaiser

**Comment by SpencerKaiser** [comment]
@johnkahn yea we chatted about it yesterday. I didn't consider the added difficulty of learning to write tests for this issue so I thought it was fair to have them focus on it later

File: src/api/supportRequest.ts [hunk 1]
```diff
@@ -12,6 +12,34 @@ export interface NextSupportRequestResponse {
...
+        requests = await SupportRequest.find({ status: SupportRequestStatus.Complete });
+        res.send(requests);
+      case "Pending":
+        requests = await SupportRequest.find({ status: SupportRequestStatus.Pending });
+        res.send(requests);
+      case "InProgress":
+        requests = await SupportRequest.find({ status: SupportRequestStatus.InProgress });
+        res.send(requests);
+      default:
+        res.send("It doesn't look like that's a valid status!");
```
Comment (SpencerKaiser) [inline src/api/supportRequest.ts:35]: It's super important with situations like this that you follow HTTP standards to communicate with the front end, so we need to update this to send a status back to the front end. In this situation, the frontend goofed up in making the request, so the status should indicate that the client was at fault. I'll let you figure out what status to use. I think the correct syntax is `res.status(XXX).send('Invalid status')`

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
A couple of quick things, but this is a great start!!

**Comment by SpencerKaiser** [comment]
Also @Ulfingrad make sure your issue description is human-readable. Go take a look at other issues for a better idea of what we're looking for, but for this I'd suggest "Add new /getAll route for support requests" or something similar.

**Review by SpencerKaiser — APPROVED** [review]
:shipit:

**Pull request merged**